scripts/inner/kubestellar

```diff
@@ -159,32 +148,58 @@ while (( $# > 0 )); do
     esac
     shift
 done
+if [ "$PROVIDER_NAME" == "" ]; then
+    PROVIDER_NAME="default"
+    export PROVIDER_NAME
+fi
+if [ "$PROVIDER_NAMESPACE" == "" ]; then
+    PROVIDER_NAMESPACE=spaceprovider-${PROVIDER_NAME}
+    export PROVIDER_NAMESPACE
+fi
+if [ "$KUBECONFIG_DIR" == "" ]; then
+    KUBECONFIG_DIR="${PWD}/temp-space-config"
+    mkdir -p $KUBECONFIG_DIR
+    export KUBECONFIG_DIR
+fi
+if [ "$SPACE_MANAGER_KUBECONFIG" == "" ]; then
+    SPACE_MANAGER_KUBECONFIG="${KUBECONFIG_DIR}/config"
+    export SPACE_MANAGER_KUBECONFIG
+fi
+if [ "$IN_CLUSTER" == "" ]; then
+    IN_CLUSTER=true
+    export IN_CLUSTER
+fi
+echo "KUBECONFIG_DIR=${KUBECONFIG_DIR}"
+echo "SPACE_MANAGER_KUBECONFIG=${SPACE_MANAGER_KUBECONFIG}"
+echo "PROVIDER_NAME=${PROVIDER_NAME}"
+echo "PROVIDER_NAMESPACE=${PROVIDER_NAMESPACE}"
+echo "IN_CLUSTER=${IN_CLUSTER}"
 
 if [ "$subcommand" == "" ]; then
     echo "$0: missing a subcommand" >&2
     exit 1
 fi
 
-case "$local_kcp" in
-    (true|false) ;;
-    (*) echo "$0: --local-kcp must be given 'true' or 'false', not '${local_kcp}'" >&2
-	exit 1;;
-esac
-
-
-
 function create_or_replace() { # usage: filename
     filename="$1"
     kind=$(grep kind: "$filename" | head -1 | awk '{ print $2 }')
     name=$(grep name: "$filename" | head -1 | awk '{ print $2 }')
-    if kubectl get "$kind" "$name" &> /dev/null
+    if kubectl get "$kind" "$name" # &> /dev/null
     then eval kubectl replace -f "$filename" $verbdir
     else eval kubectl create -f "$filename" $verbdir
     fi
 }
 
-# current ws at start does not matter, is root:compute on return
 function ensure_root_compute_configd() {
+    # TODO: remove kcp dependency in PR3 once we switch to kube-bind.
+    secret_name=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get spaceproviderdesc $PROVIDER_NAME -o jsonpath='{$.spec.secretRef.name}'`
+    secret_namespace=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get spaceproviderdesc $PROVIDER_NAME -o jsonpath='{$.spec.secretRef.namespace}'`
+    echo "secret_name=${secret_name}"
+    echo "secret_namespace=${secret_namespace}"
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get secret $secret_name -n $secret_namespace -o jsonpath='{$.data.kubeconfig}' | base64 -d > ${KUBECONFIG_DIR}/kcp.kubeconfig
+    KUBECONFIG=${KUBECONFIG_DIR}/kcp.kubeconfig
```

Comment:
This assignment has non-local effects (beyond the execution of `ensure_root_compute_configd`). I think it would be better to not have this effect. You could define this function using parens rather than curly braces around its body, or use the `local` declaration.